### PR TITLE
Add an e2e test for AdvancedAuditing

### PR DESCRIPTION
**Pull request by timstclair**
Enable a simple "advanced auditing" setup for e2e tests running on GCE, and add an e2e test that creates & deletes a pod, a secret, and verifies that they're audited.

Includes https://github.com/kubernetes/kubernetes/pull/46548

For https://github.com/kubernetes/features/issues/22

/cc @ericchiang @sttts @soltysh @ihmccreery

### CODE REVIEW

File: test/e2e/audit.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.KubeDescribe("Audit", func() {
+	f := framework.NewDefaultFramework("audit")
+
+	It("should audit API calls", func() {
+		if !strings.Contains(framework.TestContext, "AdvancedAuditing=true") {
```
Comment (liggitt) [inline test/e2e/audit.go]: TestContext isn't a string, right? How does this work?

Comment (timstclair) [inline test/e2e/audit.go]: Oops, forgot to commit local change...

File: staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go [hunk 2]
```diff
@@ -71,6 +71,7 @@ func WithAudit(handler http.Handler, requestContextMapper request.RequestContext
 		if level == auditinternal.LevelNone {
 			// Don't audit.
 			handler.ServeHTTP(w, req)
+			return
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go]: :+1:

File: test/e2e/audit.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.KubeDescribe("Audit", func() {
+	f := framework.NewDefaultFramework("audit")
+
+	It("should audit API calls", func() {
+		if !strings.Contains(framework.TestContext.FeatureGates, "AdvancedAuditing=true") {
```
Comment (sttts) [inline test/e2e/audit.go]: can we check whether logging is on?

Comment (timstclair) [inline test/e2e/audit.go]: Unfortunately no, and it turns out that even checking the feature gates doesn't worked. I moved the test to a be a `Feature` test for now, which means it needs to be explicitly included in certain test jobs.

Comment (sttts) [inline test/e2e/audit.go]: You could grep the kube-apiserver.log. Yes, ugly. But maybe the best we can do.

**Comment by liggitt** [comment]
/approve

I squinted at the salt stuff, but would like a second from someone more familiar with it

is the e2e active in any current CI envs?

**Comment by timstclair** [comment]
> is the e2e active in any current CI envs?

No. I'll follow up with that. Or is the preferred order to enable it first, and then add the test?

**Comment by liggitt** [comment]
> No. I'll follow up with that. Or is the preferred order to enable it first, and then add the test?

Not sure for tests like this. I just wondered if green runs meant it passed, or just that it compiled.

**Comment by sttts** [comment]
Lgtm for the non-salt change. Who can look at the salt changes?

File: cluster/gce/gci/configure-helper.sh [hunk 4]
```diff
@@ -1575,6 +1613,9 @@ if [[ "${KUBERNETES_MASTER:-}" == "true" ]]; then
   create-master-auth
   create-master-kubelet-auth
   create-master-etcd-auth
+  if [[ "${ENABLE_APISERVER_ADVANCED_AUDIT:-}" == "true" ]]; then
```
Comment (mikedanese) [inline cluster/gce/gci/configure-helper.sh]: is it possible to move this into the top of start-kube-apiserver where we setup kube-apiserver's log files?

**Comment by mikedanese** [comment]
/lgtm

one suggestion

File: cluster/gce/gci/configure-helper.sh (hunk 4 above)
Comment (timstclair) [inline cluster/gce/gci/configure-helper.sh]: Done (moved to where we setup the mount config for the policy file in `start-kube-apiserver`)

**Comment by timstclair** [comment]
Addressed simple comment, reapplying lgtm.

**Comment by timstclair** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Comment by timstclair** [comment]
Test failure due to https://github.com/kubernetes/kubernetes/issues/46713

**Comment by sttts** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Pull request merged**